### ISSUE TITLE
Uk changes

### ISSUE THREAD

**Issue by doskabouter** [openstreetmap/osm2pgsql#1566]
## What version of osm2pgsql are you using?

2021-09-02 13:16:01  osm2pgsql version 1.5.0
Build: Release
Compiled using the following library versions:
Libosmium 2.17.0
Proj [API 4] Rel. 4.9.3, 15 August 2016
Lua 5.3.4

## What operating system and PostgreSQL/PostGIS version are you using?

Windows, Postgres 11.2, Postgis 2.5

## Tell us something about your system

Vm 30 cores, 128GB Ram, but not really relevant here

## What did you do exactly?

ran an update of a day's changes with osmosis (latest version) and collected the changes that are done to https://www.openstreetmap.org/relation/6038068/

## This strikes me as odd:

The changes that I see that are done in the database (I used a trigger on the tables to populate a table with all the changes that are applied) for the UK are not what I expected.

When applying the changes for changeset for sequencenr 3227 (of the planet file, timestamp=2021-07-14T00\:00\:00Z) I see in my changes table 7 inserts (for each polygon of the 6038068 relation), followed a while later by 7 deletes for each polygon of those polygons, follocvdfwed later by 7 inserts of those same polygons again.

This really messes up my tile-invalidation scheme, based on the Insert-Delete-Insert, I technically have to invalidate all tiles that overlap that polygon.

From what I can find out there are no tag changes, only geochanges.
Also logged the times of those inserts/deletes, and it seems like the first insert was added in the "Reading input files" stage and the delete and 2nd insert were done in the "Going over pending relations" stage.

I know I'm a bit lagging behind on updates, but for the case in hand I think it's not really relevant.
Also I noticed this for a couple of days more preceding the 3227 sequence but unfortunately I didn't log enough info to deeply analyze it and present it here.
Tried to check the history of that relation on openstreetmap.org and the deep diff of the relation on http://osm.mapki.com, but both timeout before returning any results, so that doesn't help either

So to conclude, my guess would be that some relations are wrongly marked as pending and thus get processed twice.

If I have some more logging in the next couple of days, I'll analyze them and add conclusions here
Edit: checking all the 3x7 polygons, there are no changes at all in the geometry.
Can I rely on the fact that osmosis merges all changes to one item into one? (f.e. if someone adds a polygon, then deletes it, and then adds it again, only the last addition is passed to osm2pgsql?

**Comment by joto** [comment]
This looks like a duplicate of #423.

> Can I rely on the fact that osmosis merges all changes to one item into one? (f.e. if someone adds a polygon, then deletes it, and then adds it again, only the last addition is passed to osm2pgsql?

Well, that's really something you have to ask Osmosis, not us. :-) But osm2pgsql will only work if it gets each object only once in its input file, otherwise it will complain.

**Comment by doskabouter** [comment]
Ok, thanks!

Indeed I'm using --simplify-change and according to the docs that would merge changes into one.
Seeing that others noticed it too and had a bit deeper investigations/discussion, combined with your remark about duplicate objects strengthens my belief that in this case I can fully focus on the first delete and last insert to find out what exactly changed.
Still odd that nothing (geo nor tags) seems to be changed in my case.
